src/pip/_internal/models/search_scope.py

```diff
@@ -77,11 +77,28 @@ def __init__(
     def get_formatted_locations(self):
         # type: () -> str
         lines = []
+        index_urls = []
         if self.index_urls and self.index_urls != [PyPI.simple_url]:
+            for url in self.index_urls:
+
+                # Parse the URL
+                purl = urllib_parse.urlsplit(url)
+
+                # URL is generally invalid if scheme and netlock is missing
+                # there are issues with Python and URL parsing, so this test
+                # is a bit crude. See bpo-20271, bpo-23505. Python doesn't
+                # always parse invalid URLs correctly - it should raise
+                # exceptions for malformed URLs
+                if not purl.scheme and not purl.netloc:
+                    logger.warning('index-url {} is invalid.'.format(url))
+                    continue
```

Comment:
I woud suggest still keeping the invalid URL, to avoid false positives. pip’s URL checking logic is very lax, and I fear we might receive unnecessary complaints if we chage the behaviour here. The process will fail eventually if the URL is indeed invalid, and this extra message should be enough for a lot of people to debug the issue.

Reply:
Sure @uranusjr , I can get rid of the `continue` here to make that happen

Reply:
Hi @uranusjr , 

I have made the requested changes. Please take a look :)